ntoskrnl/ke/eventobj.c

```diff
@@ -37,6 +37,7 @@ KeInitializeEvent(OUT PKEVENT Event,
                   IN BOOLEAN State)
 {
     /* Initialize the Dispatcher Header */
+    ASSERT((Type == NotificationEvent) || (Type == SynchronizationEvent));
```

Comment:
A similar check needs to be done in `KeInitializeTimerEx` as well:
https://github.com/reactos/reactos/blob/fff6d960003fe79b17d6e52875de60d3964e446b/ntoskrnl/ke/timerobj.c#L250-L252